.github/workflows/migration_safety_warning.yml

```diff
@@ -0,0 +1,41 @@
+name: "Migration Safety Check"
+
+on:
+  pull_request:
+    paths:
+      - "**/migrations/**/*"
+
+jobs:
+  check-and-apply-labels:
+    name: "Check and Apply Labels"
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: peter-evans/find-comment@v2
+        id: existing-warning
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-includes: This PR has migrations
+
+      - uses: actions/github-script@v6
+        if: steps.existing-warning.outputs.comment-id != 0
+        with:
+          github-token: ${{ secrets.ACCESS_TOKEN }}
+          script: |
+            await github.rest.issues.deleteComment({
+              owner: 'WordPress',
+              repo: 'openverse-api',
+              comment_id: ${{ steps.existing-warning.outputs.comment-id }},
+            })
+
+      - uses: peter-evans/create-or-update-comment@v2
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body: |
+            This PR has migrations. Please rebase it before merging to ensure that conflicting migrations are not introduced.
```

Comment:
If I'm understanding this correctly, will this issue a comment on every PR that has a change to migrations, even if it's already rebased onto main?

Reply:
Yes. I've tried to make much more complicated versions of this but I'd really like to just get this basic version in to mitigate the issue for now. There are much more compelling versions of this that would just run the check for you and only comment/request changes when it determines that the PR does indeed have migration conflicts (instead of just giving a gentle nudge).

The gentle nudge is good enough for now. Kudos to @dhruvkb for encouraging a good scope on this ticket.